weed/filer/elastic/v7/elastic_store.go

```diff
@@ -0,0 +1,308 @@
+package elastic
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/chrislusf/seaweedfs/weed/filer"
+	"github.com/chrislusf/seaweedfs/weed/glog"
+	"github.com/chrislusf/seaweedfs/weed/pb/filer_pb"
+	weed_util "github.com/chrislusf/seaweedfs/weed/util"
+	jsoniter "github.com/json-iterator/go"
+	elastic "github.com/olivere/elastic/v7"
+)
+
+var (
+	indexType   = "_doc"
+	indexPrefix = ".seaweedfs_"
+)
+
+type ESEntry struct {
+	ParentId string `json:"ParentId"`
+	Entry    *filer.Entry
+}
+
+func init() {
+	filer.Stores = append(filer.Stores, &ElasticStore{})
+}
+
+type ElasticStore struct {
+	client      *elastic.Client
+	maxPageSize int
+}
+
+func (store *ElasticStore) GetName() string {
+	return "elastic7"
+}
+func (store *ElasticStore) Initialize(configuration weed_util.Configuration, prefix string) (err error) {
+	servers := configuration.GetString(prefix + "servers")
+	if servers == "" {
+		return fmt.Errorf("error elastic endpoints.")
+	}
+	store.maxPageSize = configuration.GetInt(prefix + "index.max_result_window")
+	if store.maxPageSize <= 0 {
```

Comment:
make this optional? Default to some number high enough for most cases?

Reply:
default is 10000, coded in scaffold cfg files.
it’s ok for most cases, but increase value will get better query performance.
etc, list the bucket which contains 10 million photos.

Reply:
The instruction in scaffolded yaml file is not very clear to me.

To list 10 million photos, what is a good number? Let's say it is 1 million, so why not default to 1 million? What is the limiting factor?

Seems complicated and for experts only. Please treat it as optional here and provide a default value, instead of just return an error in line 47.

Reply:
because the value in elastic is 10000,if we query more than  10000 entries a time, es
will return errors. so default is 10000，increase value will take both filer and elastic.